backend/src/govsg/services/govsg-callback.service.ts

```diff
@@ -259,12 +352,87 @@ const parseTemplateMessageWebhook = async (
   }
 }
 
+const sendPasscodeMessage = async (
+  whatsappId: WhatsAppId,
+  clientId: WhatsAppApiClient,
+  officerName: string,
+  officerAgency: string,
+  passcode: string
+): Promise<MessageId> => {
+  const templateMessageToSend: WhatsAppTemplateMessageToSend = {
+    recipient: whatsappId,
+    apiClient: clientId,
+    templateName: 'sgc_send_passcode', // TODO: Un-hardcode this
+    params: [
+      {
+        type: 'text',
+        text: officerName,
+      },
+      {
+        type: 'text',
+        text: officerAgency,
+      },
+      {
+        type: 'text',
+        text: passcode,
+      },
+    ],
+    language: WhatsAppLanguages.english,
+  }
+  const isLocal = config.get('env') === 'development'
+  const passcodeCreationWamid =
+    await WhatsAppService.whatsappClient.sendTemplateMessage(
+      templateMessageToSend,
+      isLocal
+    )
+  return passcodeCreationWamid
+}
+
 const parseUserMessageWebhook = async (
   body: UserMessageWebhook,
   clientId: WhatsAppApiClient
 ): Promise<void> => {
   const { wa_id: whatsappId } = body.contacts[0]
   const { id: messageId, type } = body.messages[0]
+  if (type === WhatsappWebhookMessageType.button) {
+    const message = body.messages[0] as WhatsAppWebhookButtonMessage
+    if (message.button.text === 'Create passcode') {
+      const passcodeCreationWamid = message.context.id
+      const govsgVerification = await GovsgVerification.findOne({
+        where: { passcodeCreationWamid },
+        include: [GovsgMessage],
+      })
+      if (!govsgVerification) {
```

Comment:
Would this be considered an erroneous state? Do we need to explicitly throw an error?

Reply:
I suppose this could be erroneous because it means somehow someone managed to tap on a `Create passcode` button when `GovsgVerification` had no record of the message.

This can happen when:

- (Erroneous) We failed to create `GovsgVerification` after sending the passcode creation message.
- (Mischief...? But I guess it's still erroneous?) Someone sent the passcode creation message or a message containing a `Create passcode` button outside of Postman's ecosystem.

Let me change `logger.warn` to `logger.error`.